### PR TITLE
MONGOID-5407 Add phantom custom field type example


### PR DESCRIPTION
**Pull request by p-mongo**

### CODE REVIEW

**Comment by johnnyshields** [comment]
How about `#mongoize` / `#demongoize` defined on object instance level? In this example, would Mongoid ever attempt to call `"black".mongoize` (i.e. instead of `ColorMapping.mongoize("black")`) and yield unexpected behavior?

Unless we have tested everything very thoroughly here and are confident the above is not a concern, I would suggest to hold off on this until [MONGOID-5409](https://jira.mongodb.org/browse/MONGOID-5409) is done. After MONGOID-5409, "Phantom Custom Field Types" will become the standard case (i.e. "Custom Field Adapters"), so if that ticket is coming soon it may also make sense to defer this.

**Comment by p-mongo** [comment]
https://jira.mongodb.org/browse/MONGOID-5407

**Comment by p-mongo** [comment]
@johnnyshields I'm afraid you misunderstood the purpose of this PR and the corresponding ticket.

The documentation added here documents existing behavior of Mongoid and is intended to help with figuring out what to do in the future, such as whether to implement 5409 and if so how.

File: docs/reference/fields.txt
Comment (p-mongo) [inline docs/reference/fields.txt:1018]:
```suggestion
    INVERSE_MAPPING = MAPPING.invert.freeze
```

File: docs/reference/fields.txt
Comment (p-mongo) [inline docs/reference/fields.txt:1022]:
```suggestion
      # Takes application-scope value and converts it to how it would be
```

File: docs/reference/fields.txt
Comment (p-mongo) [inline docs/reference/fields.txt:1029]:
```suggestion
      # value. Converts invalid values to nil.
```

**Comment by johnnyshields** [comment]
@p-mongo it's fine, and I'm all for the direction this is headed. My concern is simply that if this documentation is released in Mongoid 8.0, then users will expect this functionality to be "ready for prime time" without caveats/edge-case bugs. If not sure, better not to release. That's all.

**Comment by johnnyshields** [comment]
@p-mongo please refer to my original comment:

> How about #mongoize / #demongoize defined on object instance level? In this example, would Mongoid ever inadvertently attempt to call "black".mongoize (i.e. instead of ColorMapping.mongoize("black")) and yield unexpected behavior?

I am asking specifically "have you tested this case"? If you have not, the given the complexity of Mongoid, you and I should assume there are edge cases there. The responsible thing to do would be to test this case and others thoroughly before releasing documentation to users.

**Comment by p-mongo** [comment]
This PR includes execution results of the pattern being demonstrated. If you want to do something else please create a  ticket for your use case.

**Comment by johnnyshields** [comment]
Yes, you are testing one usage pattern narrowly, and that's a _great_ first step. However, I do not believe that based on this single "happy-path" test case alone that you should advertise to Mongoid users that this usage pattern is "ready for prime time".

The prudent thing to do would be to wait until detailed Rspec tests are added as part of [MONGOID-5409](https://jira.mongodb.org/browse/MONGOID-5409) before releasing this documentation to users. If MONGOID-5409 is certain to come in the next release (8.0), then no issue here.

Anyway, I've said my piece. It's your call.

**Review by Neilshweky — APPROVED** [review]

**Pull request merged**
